client/dashboard/data/site.ts

```diff
@@ -112,10 +113,17 @@ export interface Site {
 }
 
 export async function fetchSite( siteIdOrSlug: number | string ): Promise< Site > {
-	return await wpcom.req.get(
-		{ path: `/sites/${ siteIdOrSlug }` },
-		{ fields: JOINED_SITE_FIELDS, options: JOINED_SITE_OPTIONS }
-	);
+	try {
+		return await wpcom.req.get(
+			{ path: `/sites/${ siteIdOrSlug }` },
+			{ fields: JOINED_SITE_FIELDS, options: JOINED_SITE_OPTIONS }
+		);
+	} catch ( error ) {
+		if ( error instanceof Error && error.message.includes( '[-32710]' ) ) {
+			throw new DashboardDataError( 'inaccessible_jetpack', error );
+		}
+		throw error;
```

Comment:
My thinking here is that the error handling is fine tuned to what v2 needs. Hence why it is a *dashboard* data error. We should only feel we need to detect the error situations we care about.